### DISCUSSION TITLE
Ignore folders by default?

### DISCUSSION THREAD

**Discussion by TheRealFlyingCoder** [kiliman/remix-flat-routes#34] [Q&A]
So this might be another issue with doc clarity, but in a Hybrid folder setup we are already foregoing any "clarity at a glance" on the routing structure for the benefit of colocation so in this case I would just prefer to have the folders follow a familiar naming convention like the following:

```
/routes
  /Auth
    /Login
      /components
        /myComponent.tsx
      /login.route.tsx
```

And output the result:

```
<Routes>
  <Route file="root.tsx">
    <Route index file="routes\\index.tsx" />
    <Route path="login" file="routes\Auth\\Login\\login.route.tsx" />
  </Route>
</Routes>
```

Given we have the `+` suffix to add a folder name to the routing this should be possible but it doesn't seem that any combination works?

Essentially saying "Ignore all folders unless that have a specific suffix or prefix, and only look for *.route.tsx files"

**Comment by TheRealFlyingCoder** [comment]
i'm very aware this kind of bypasses the whole point of "flat" <-- but this will kind of become the go-to tool for every alternative routing convention in remix i'd say :P

**Reply by kiliman** [reply]
Thanks for the suggestion. I guess we can add some config options. My main concern is whether it will end up being too complex.

Ryan doesn't plan on adding too many options to the new v2 routing convention, so this package will continue to exist. It'll be a way to try out new ideas.

**Reply by TheRealFlyingCoder** [reply]
I mean this in the nicest way, but I think it's already a little complex with multiple variations of prefixes and suffixes 😅 

It might as well have all the route configuration options you'd want available at this point don't you think?

I have a feeling new users are most likely to find either pure flat routes familiar from other frameworks, or this suggestion from general development practices

**Reply by TheRealFlyingCoder** [reply]
Okay so a follow up to this, I did find a solution that i'm happy with in the current version.

It's closer to a V1 route setup but lets me co-locate any files as long as they don't end in .route or start with _. 

```
/routes
  /_auth
    /login
      /components
        /myComponent.tsx
      /_index.tsx   <-- Headers / Meta / and exports the others
      /Login.tsx <-- Markup
      /Login.loader.tsx
      /Login.action.tsx
```

This is similar to how I originally did my file setup (splitting of actions/loaders etc to maintain focus), but those files would be in a `~/views` folder, this just brings them all together and works fine.

I'm really finding it hard to choose what i'd prefer though, we are moving into territory now that it's unclear where the route lands at a glance, but with `npx remix routes` I guess it's not that big a concern?

**Reply by kiliman** [reply]
Now that the v2 convention is out, I want to sit back and rethink this. We do a lot more than just flat routes, and Ryan says he's most likely not going to include any of the enhancements in Remix core.

Perhaps something like a _routing convention construction kit_, where you can define what is considered a route, a parent route, etc. And we have some pre-configured kits to emulate v1 routing, v2 routing, etc.
